### DISCUSSION TITLE
Hope someday there are UI Widgets in C# Drawings and update, like Flutter no more XAML.

### DISCUSSION THREAD

**Discussion by nextcodelab** [dotnet/maui#12832] [General]
I'm am developing .net maui several months,, but very lags in everything maybe because xaml need lots of compilations than C# itself and it consume from native UI (THIS NEED LOTS OF CODE AND LOGICS).
Hope there are teams that focuses on drawings UI, update draw (Stateful and Stateless Pure C#) every UI should have different purpose like 
SIZEBOX,
ROW,
COLUMN, 
PADDING,
CENTER,
CARD,
ICON.
ICONBUTON,
Different kind of buttons,
DEVIDER,
TEXT,
Different UI for every specific purpose, to prevent lots of properties, lots of codes, lots compilations and logics in every UI.

I feel that flutter surpass even WINUI in fastest HOT RELOAD, HOT RESTART is just a millisecond in flutter.

Now I tried flutter and I feel no lags even a single spot of lags,, the start up in flutter is just 1 second in Windows compare to WinUI,
Since Flutter is everything is a widgets, you can insert any widget contents as widgets, so everything is customizable,
![Screenshot 2023-01-23 191550](https://user-images.githubusercontent.com/73583588/214027335-e3d9bcad-e9e3-43cc-b1d9-58f2e993aea5.png)

FLUTTER APP IN RELEASE MODE CAN ACHIEVE 1 SECOND IN START-UP, BECAUSE NO MORE LOTS OF PROPERTIES AND LOGIC LIKE XAML.

https://user-images.githubusercontent.com/73583588/215910248-f0f69dcb-e4d3-4527-9ea7-64587c04e9e8.mp4

No more delays in release mode. 
https://user-images.githubusercontent.com/73583588/216351217-807a6ec3-151f-4a3d-9cc5-cb7333f6213c.mp4

**Comment by saint4eva** [comment]
Kindly use any of the following amazing C# only UI libraries:

- [Comet](https://github.com/dotnet/Comet)
- [Sharp.UI](https://github.com/idexus/Sharp.UI/)
- [C# Markup](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/markup/markup)
- [Spice UI](https://github.com/jonathanpeppers/spice)

**Reply by nextcodelab** [reply]
Sharp UI still use .Net Maui  Controls I feel the lags, sorry for the words,, this is what I feel in .Net Maui NATIVE Bridge?

**Reply by saint4eva** [reply]
What do you mean by native? Do you mean .NET for Android and .NET for iOS? If not, Maui controls render natively on each platform - be it Android, iOS or Windows.

However, Sharp.UI is more of mvvm pattern, while Comet and ReactorUI are more of MVU pattern

**Reply by saint4eva** [reply]
But since you are interested in flutter, there is another beautiful C# library that uses drawn UIs - [AlohaKit UI](https://github.com/jsuarezruiz/AlohaKit.UI)

**Reply by nextcodelab** [reply]
Maybe this works, we will try the start up time... and instance draw in RELEASE MODE.

**Reply by saint4eva** [reply]
> Maybe this works, we will try the start up time... and instance draw in RELEASE MODE.

Can't wait to see your result.

**Comment by artemvalieiev** [comment]
You can achieve less than 1 ms startup time in release mode, see https://devblogs.microsoft.com/dotnet/performance-improvements-in-dotnet-maui

**Comment by Xyncgas** [comment]
I dissagree that the future of UI framework is a bunch of control, most of them are going to be the same anyways, just different layout, a button it's just a button it's not an image, and it can even not be a button but an element with onclick event

you should only use other control when you need to do something you couldn't, otherwise just draw your own 'card' and 'row' with absolute positioning and child elements, you would appreciate it later when you want to do something that microsoft didn't do for you

write your own algorithms for calculating UI, (put x in the center of y |> zoom 0.5f |> putCornerAt (BL, y.BL)), it might actually be faster most importantly now you just need the rendering engine to do exactly what you are telling it to draw and call update to redraw and you have complete control over how and the performance of your UI because you wrote the algorithms to calculate them you wrote the code to put objects to heap you wrote the codes the dispatch a captured image to multiple Renderer for shared-grain parallelism without all the complexity microsoft shipped in their XAML black box that you don't entirely know at first because you didn't write them

how else are you going to render multiple part of the screen in parallel with single threaded UI in Xamarin

Meanwhile this is very simple to implement, because I don't want you to ship a bunch of control, I just need a bug free framework that can draw my description of the UI to the screen without using Xaml. So I can stick with one element for most of the thing like 'div' in browser, XAML can retire because there isn't anything to declare anyways with vast amount of controls reduced, and having to write the tag name twice sucks (IDE has auto complete ok), and we can even have anonymous element description which you can't do with XAML in actual programming languages, we can all keep things very simple and be happy

**Comment by davepruitt** [comment]
While I do like Flutter, I much prefer using XAML for my UI development. Flutter code can very quickly get messy and "hilly". It's difficult to understand what is going on in Flutter UI code. XAML is much more easily comprehended.
